plugins/versionpress/src/Cli/vp-composer.php

```diff
@@ -0,0 +1,179 @@
+            array_map($this->createPackageToChangeInfoMapFunction('install', $plugins, $themes), $installedPackages),
+            array_map($this->createPackageToChangeInfoMapFunction('delete', $plugins, $themes), $removedPackages),
+            array_map($this->createPackageToChangeInfoMapFunction('update', $plugins, $themes), $updatedPackages)
+        )));
+
+        foreach ($changeInfos as $changeInfo) {
+            $committer->forceChangeInfo($changeInfo);
+        }
+
+        $committer->commit();
+    }
+
+    private function detectChanges()
+    {
+        $currentComposerLock = file_get_contents(VP_PROJECT_ROOT . '/composer.lock');
+
+        $process = new Process(VP_GIT_BINARY . ' show HEAD:composer.lock', VP_PROJECT_ROOT);
+        $process->run();
+
+        $previousComposerLock = $process->getOutput();
+
+        $currentPackages = $this->getPackagesFromLockFile($currentComposerLock);
+        $previousPackages = $this->getPackagesFromLockFile($previousComposerLock);
+
+        $installedPackages = array_diff_key($currentPackages, $previousPackages);
+        $removedPackages = array_diff_key($previousPackages, $currentPackages);
+
+        $packagesWithChangedVersion = array_filter(
+            array_intersect_key($currentPackages, $previousPackages),
+            function ($package) use ($previousPackages) {
+                return $package['version'] !== $previousPackages[$package['name']]['version'];
+            }
+        );
+
+        return [
+            'installed' => $installedPackages,
+            'removed' => $removedPackages,
+            'updated' => $packagesWithChangedVersion,
+        ];
+    }
+
+    private function getPackagesFromLockFile($lockFileContent)
+    {
+        $lockFile = json_decode($lockFileContent, true);
+        return array_combine(
+            array_column($lockFile['packages'], 'name'),
+            array_map(function ($package) {
+                return [
+                    'name' => $package['name'],
+                    'version' => $package['version'],
+                    'type' => $package['type'],
+                    'homepage' => @$package['homepage'] ?: null,
+                ];
+            }, $lockFile['packages'])
+        );
+    }
+
+    private function getPackages()
+    {
+        return ['plugins' => get_plugins(), 'themes' => wp_get_themes()];
+    }
+
+    private function createPluginChangeInfo($action, $fullPackageName, $plugins)
+    {
+        list($vendor, $packageName) = explode('/', $fullPackageName);
+
+        foreach ($plugins as $pluginFile => $plugin) {
+            list($pluginDirectory) = explode('/', $pluginFile, 2);
+
+            if ($packageName === $pluginDirectory) {
+                return new PluginChangeInfo($pluginFile, $action, $plugin['Name']);
+            }
+        }
+
+        return null;
+    }
+
+    private function createThemeChangeInfo($action, $fullPackageName, $themes)
```

Comment:
Shouldn't this method be placed to some kind of factory, or as static to `ThameChangeInfo.php` ? I have same question for the method `createPluginChangeInfo`.